File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.Arrays;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        int[] recode = new int[k];
        Arrays.fill(recode,-1);
        for(int i=0; i<score.length; i++){
            if(recode[0]<score[i]){
                recode[0] = score[i];
            }
            Arrays.sort(recode);
            if(i<k-1){
                answer[i] = recode[k-i-1];
            }else{
                answer[i] = recode[0];
            }
        }
        
        
        return answer;
    }
}

File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    var answer = [];
    //stack
    var stack = [];
    score.forEach(e => {
        if(stack.length===0) stack.push(e);
        else{
            var pop = stack.pop();
            var max = Math.max(e, pop);
            var min = Math.min(e, pop);
            stack.push(max);
            if(stack.length < k) stack.push(min);
        }
        stack.sort((a,b) => b-a);
        answer.push(stack[stack.length-1]);
    })
    return answer;
}